config/initializers/rails_performance.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_performance'
+
+Rails.application.config.after_initialize do
+  if Settings.rails_performance.enabled
+    RailsPerformance::RailsPerformanceController.class_eval do
+      include SessionHandler
+      before_action :check_user_not_authorized
+
+      # Modify CSP for Performance Gem
+      content_security_policy do |p|
+        p.style_src :self, "'unsafe-inline'"
+        p.script_src_elem :self,
+                          "'unsafe-inline'",
+                          'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/js/all.min.js',
+                          'https://code.highcharts.com'
+      end
+
+      protected
+
+      # Helper to use SessionHandler to check for permissions
+      def check_user_not_authorized
+        unless real_user&.admin_user?
+          render 'shared/http_status',
+                 formats: [:html], locals: { code: '403', message: HttpStatusHelper::ERROR_CODE['message']['403'] },
+                 status: :forbidden, layout: false
+        end
+      end
+    end
+  end
+
+  if defined?(RailsPerformance)
+    RailsPerformance.setup do |config|
+      config.enabled = Settings.rails_performance.enabled
+      config.redis = Redis::Namespace.new(Rails.root.to_s)
+      config.duration = Settings.rails_performance.duration.minutes
+      config.mount_at = '/admin/performance'
```

Comment:
The [rails_performance documentation](https://github.com/igorkasyanchuk/rails_performance#usage) seems to indicate that we can omit this setting if we manually mount the server in our `routes.rb` file.

Reply:
You can. This was just added for redundancy. I will remove it.

Reply:
Actually after testing it a bit more, I have found that it can be omitted, though I had to change the mounting point to the default `/rails/performance` if thats alright.

I found if I changed the mount point for the gem like this:
```ruby
namespace :admin do
  ...
  mount RailsPerformance::Engine, at: '/performance', as: 'performance'
end
```
the links for each tab in the gem redirects to `<root_path>/rails/performance/<tab_name>` and are thus broken.

I think this has something to do with the admin namespace because when I move `mount RailsPerformance::Engine, at: '/performance', as: 'performance'` outside the admin namespace and rename it as `mount RailsPerformance::Engine, at: 'admin/performance', as: 'performance'` the gem works fine again.